### ISSUE TITLE
Test + docs: dataset/preprocess

### ISSUE THREAD

**Issue by NMontanaBrown** [DeepRegNet/DeepReg#100]
# Issue description

Unit tests + documentation for 2 functions in deepreg/dataset/preprocess

- AffineTransformation3D
- preprocess: dependent on AffineTransformation3D

This is heavily dependent on layer_utils - issue #55. So we need to resolve ticket #55 before we tackle this, likely

## Type of Issue

Please delete options that are not relevant.
- [x] Documentation update
- [x] Test request
- [x] Linting request

#### What's the expected result?
Unit tests for outputs, types, shapes. Linting requests, docstrings and docs updates.

**Comment by mathpluscode** [comment]
@zacbaum please be aware of the coming PR #150. Where affine transformation is not touched but we added a new function `resize_inputs`, the corresponding tests are added in `test_preprocess_resize.py`. I put it in a different file to prevent merge conflicts for you.

I see your branch seems to have all the tests. Once #150 is merged into the master branch, please synchronise and move the test of resize into your `test_preprocess.py`.

**Comment by zacbaum** [comment]
Thanks for the heads up @mathpluscode, will do before adding to the PR you've created (which, in the future, I am more than capable of doing myself...) for this ticket once all the work is finished.

**Comment by mathpluscode** [comment]
@zacbaum it's merged ;) please ping me in case of any problems!

**Comment by NMontanaBrown** [comment]
So what is the situation here? I am confused as to whether this has been merged or not by @mathpluscode when it was not finished? does it require extra work? are there new functionalities? @mathpluscode  could you clarify?

**Comment by zacbaum** [comment]
@mathpluscode added some new functionality (and tests) which will be integrated in a future commit to the branch for this, but currently, we need to decide if we want to test anything further than what is there now (will make a new commit shortly that can be looked at).

I think the main confusion was that `preprocess` was really only used in `interface.py` but that has since been moved and will be tested elsewhere... So this should be good to go once I integrate the changes from @mathpluscode.

**Comment by mathpluscode** [comment]
@NMontanaBrown  so, it's just i added a function `resize_inputs` into `preprocess.py` and i removed a function from it.

I provided the tests for the new function and @zacbaum has moved it into his test script.

In conclusion, no problems here and we are good ;)
